erc20/lib.rs

```diff
@@ -178,6 +178,8 @@ mod erc20 {
                 return Err(Error::InsufficientAllowance)
             }
             self.transfer_from_to(&from, &to, value)?;
+            // We checked that allowance >= value
+            #[allow(clippy::arithmetic_side_effects)]
```

Comment:
A more idiomatic way would be to have use checked subtraction above instead and throw an error on `None` instead of suppressing and adding comment here